src/retry-strategy/exponential-backoff-retry-strategy.ts

```diff
@@ -0,0 +1,196 @@
+import { NetworkRequest } from '../request';
+import { BaseSubscription } from '../subscription/base-subscription';
+import { ErrorResponse, NetworkError } from '../base-client';
+import { UnauthenticatedRetryStrategy } from './unauthenticated-retry-strategy';
+import { EmptyLogger, Logger } from '../logger';
+import { DoNotRetry, Retry, RetryStrategy, RetryStrategyResult } from './retry-strategy';
+
+export interface ExponentialBackoffRetryStrategyOptions {
+    tokenFetchingRetryStrategy?: RetryStrategy, //Retry strategy that checks for expired token and fetches it
```

Comment:
Could this just be a token provider? Whether it retries or not should be token provider's implementation detail.